dtfs-central-api/src/v1/controllers/portal/gef-deal/add-min-activities.controller.js

```diff
@@ -84,25 +84,26 @@ const facilityChangePortalActivity = async (application, facilities) => {
     const { checkerId, portalActivities } = application;
     const checker = await getUserInfo(checkerId);
 
-    facilities.forEach(async (facility) => {
-      if (facility.canResubmitIssuedFacilities) {
-        // creates user object to add to array
-        const maker = facility.unissuedToIssuedByMaker;
-        const activityParams = {
-          type: PORTAL_ACTIVITY_LABEL.FACILITY_CHANGED_ISSUED,
-          user: '',
-          activityType: PORTAL_ACTIVITY_TYPE.FACILITY_STAGE,
-          activityText: '',
-          activityHTML: 'facility',
-          facility,
-          maker,
-          checker,
-        };
-        // generates an activities object
-        const activityObj = portalActivityGenerator(activityParams);
-        // adds to beginning of portalActivities array so most recent displayed first
-        portalActivities.unshift(activityObj);
+    facilities.forEach((facility) => {
+      if (!facility.canResubmitIssuedFacilities) {
+        return;
```

Comment:
Is this required? Otherwise the one liner is simpler?:

```js
if (facility.canResubmitIssuedFacilities) { ...
```

Reply:
I don't think this is necessarily simpler. If I remember correctly, this change was made because guard clauses are often easier to read than branching code - the early `return` here if the condition is met makes it clear to the reader that, if you can't resubmit issued facilities, then you do nothing. In contrast, a non-guard clause approach means that a reader would read this code as "if you can resubmit issued facilities, perform this action", which means they need to scroll to the bottom of the `if` block to figure out what happens if you can't resubmit facilities

Reply:
Thanks @JoshBinns2000 that make sense RE readability. Happy with either approach.